recognition/include/pcl/recognition/dense_quantized_multi_mod_template.h

```diff
@@ -66,10 +67,21 @@ namespace pcl
 
       std::size_t num_of_features;
       read (stream, num_of_features);
-      features.resize (num_of_features);
-      for (std::size_t feature_index = 0; feature_index < num_of_features; ++feature_index)
+      if (stream.bad())
```

Comment:
Maybe create a function for this. The function takes in a stream, checks and throws. You've used this snippet multiple times. It'll look like this:
```cpp
read (stream, num_of_features);
throw_if_io_fail(stream);
// no need of error checking since throw is propagated
for (std::size_t ...
```

Reply:
Yeah I thought this would be better too, that's why I had mentioned it in a comment on the issue. Should I implement this function in common/include/pcl/exceptions.h ?

Reply:
You could add an optional second string parameter for the error message to be printed?